internal/test/assert.go

```diff
@@ -0,0 +1,130 @@
+package test
+
+import (
+	"bytes"
+	"io"
+	"net/http"
+	"strings"
+	"testing"
+	"time"
+
+	gokong "github.com/kong/go-kong/kong"
+	"github.com/stretchr/testify/require"
+)
+
+type configurationOption struct {
+	responseChecker []func(*testing.T, *http.Response) bool
+	waitFor         time.Duration
+	interval        time.Duration
+}
+
+type ConfigurationOpt func(*configurationOption)
```

Comment:
Nit: this name might still be a bit vague for the user. How about something along the lines of `HTTPResponseExpectOpt`?